### ISSUE TITLE
PESQ value for normalized (predicted & target) and denormalized (predicted & target) form differs

### ISSUE THREAD

**Issue by itz-vp** [Lightning-AI/metrics#1174]
## 🐛 Bug

Hi, I have tried to calculate metrics for my predicted output and target audio. 

- Calculated PESQ with normalized input
- Calculated PESQ with de normalized input

I expect both values to be same but values differ. Only for few audios am getting a huge difference, for most of the audios it is negligible difference.

#### Code sample

Given normalized audio array
---------------------------------------

![image](https://user-images.githubusercontent.com/100348383/183024175-5ee33480-0502-4080-bd0c-b6b2039b3130.png)

- Resampling the array to 16khz

![image](https://user-images.githubusercontent.com/100348383/183024298-88cbc6f0-8311-4d3a-aa65-c517854647de.png)

- PESQ Calculation

![image](https://user-images.githubusercontent.com/100348383/183024389-2457db36-2a44-425e-8361-eeb12a6b4c58.png)

Given a denormalized audio array
--------------------------------------------

![image](https://user-images.githubusercontent.com/100348383/183024532-884baf23-1d9b-456c-b476-bcfd725d496f.png)

- Resampling the array to 16khz

![image](https://user-images.githubusercontent.com/100348383/183024638-7628f237-db45-420d-ba7d-cd4ba8cb0ced.png)

- PESQ Calculation
![image](https://user-images.githubusercontent.com/100348383/183024733-5be7b40c-ea8b-4292-ab9b-896968117ea3.png)

PESQ value differs for both the normalized input and denormalized input.

What should be the input to calculate PESQ value, `A normalized version or denormalized version ?` Thanks in advance

### Expected behavior

Expecting a same PESQ value

### Environment

# Name                    Version                 
pytorch-lightning  -         1.5.10                
torch                -     1.11.0+cu115              
torchaudio         -       0.11.0+cu115            
torchmetrics         -     0.8.2                    

OS
----
 Operating System: Ubuntu 20.04.2
 Kernel: Linux 5.15.0-43-generic
Architecture: x86-64

**Comment by SkafteNicki** [comment]
cc: @quancs

**Comment by quancs** [comment]
@itz-vp  Hello, could you please upload the preds and targets (npz file) you found that have different pesq values.

**Comment by quancs** [comment]
> What should be the input to calculate PESQ value, A normalized version or denormalized version ? Thanks in advance

The following is from [torchaudio document](https://pytorch.org/audio/stable/backend.html#torchaudio.backend.sox_io_backend.load). As it says, the normalize argument doesn't change the value of what you load, it only changes your datatype to float32. Thus, it doesn't matter whether it is normalized or not.
> normalize argument does not perform volume normalization. It only converts the sample type to torch.float32 from the native sample type.

I suspect that your code changed the values such that normalized version and denormalized version have different values, thus the pesq return different pesq metric value. If it is possible, you can upload the wav and code (in a zip file). I can then do some further checks

**Comment by itz-vp** [comment]
> can upload the wav and code (in a zip file)

https://workdrive.zohoexternal.com/external/5faae69fe51950da6f4ceaec79a7b02b4874e560b246193a3cb2da3f37e2cbd4

**Comment by quancs** [comment]
@itz-vp I write a simple test with the `sample_29_clean_denorm.wav` you given trying to recover the problem in your code.

```python
import torchaudio
import torch

def equal(a: torch.Tensor, b: torch.Tensor):
    a_ = a / torch.max(torch.abs(a))
    b_ = b / torch.max(torch.abs(b))
    return (a_ == b_).all()

target_denormalized, targ_sample_rate = torchaudio.load("sample_29_clean_denorm.wav", normalize=False)
target_normalized, targ_sample_rate = torchaudio.load("sample_29_clean_denorm.wav", normalize=True)
print("target_denormalized=", target_denormalized)
print("target_normalized=", target_normalized)
print("target_denormalized == target_normalized", equal(target_denormalized, target_normalized))

target_denormalized_resample = torchaudio.functional.resample(target_denormalized.to(torch.float32), targ_sample_rate, 16000)
target_normalized_resample = torchaudio.functional.resample(target_normalized, targ_sample_rate, 16000)
print("target_denormalized_resample=", target_denormalized_resample)
print("target_normalized_resample=", target_normalized_resample)
print("target_denormalized_resample == target_normalized_resample", equal(target_denormalized_resample, target_normalized_resample))

target_denormalized_resample_int16 = target_denormalized_resample.to(torch.int16)
print("target_denormalized_resample_int16 == target_normalized_resample", equal(target_denormalized_resample_int16, target_normalized_resample))
print(target_denormalized_resample.max(), target_denormalized_resample.min())
```

The output is 
```python
target_denormalized= tensor([[ 0,  0,  0,  ..., -5, -4,  1]], dtype=torch.int16)
target_normalized= tensor([[ 0.0000e+00,  0.0000e+00,  0.0000e+00,  ..., -1.5259e-04,
         -1.2207e-04,  3.0518e-05]])
target_denormalized == target_normalized tensor(True)
target_denormalized_resample= tensor([[-0.0145,  0.0138, -1.2193,  ..., -6.1723, -4.7454, -1.0959]])
target_normalized_resample= tensor([[-4.4317e-07,  4.2247e-07, -3.7211e-05,  ..., -1.8836e-04,
         -1.4482e-04, -3.3443e-05]])
target_denormalized_resample == target_normalized_resample tensor(True)
target_denormalized_resample_int16 == target_normalized_resample tensor(False)
```

From the output, it is easy to confirm that the `.to(torch.int16)` you applied on a `float32` Tensor makes the tensor lost its original information. So, as I mentioned before, you code indeed changes the values of the wavs, making them not equal to the original wavs in the file.
